src/fetchers/candles.tsx

```diff
@@ -0,0 +1,22 @@
+import { useQuery } from '@tanstack/react-query';
+import { CandlestickData } from '@penumbra-zone/protobuf/penumbra/core/component/dex/v1/dex_pb';
+
+export const useCandles = (
+  symbol1: string,
+  symbol2: string,
+  startBlock: number | undefined,
+  limit: number,
+) => {
+  return useQuery({
+    queryKey: ['candles', startBlock],
```

Comment:
question: is this key array enough to cache correctly?

Asking cause I'm just learning react-query and truly want to understand its caching mechanisms. The [docs mention](https://tanstack.com/query/latest/docs/framework/react/guides/query-keys#array-keys-with-variables) a serializable object with key parameters like `useQuery({ queryKey: ['todo', 5, { preview: true }], ...})` that help identifying the correct query. Should it be smt like this?

```suggestion
    queryKey: ['candles', { symbol1, symbol2, startBlock, limit }],
```

Also, @grod220 asked about the server caching before. According to [NextJS docs](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching#fetching-data-on-the-server-with-the-fetch-api), using the `fetch` API should result in correct caching by the framework. Maybe we don't even need to set up the cache by react-query since it would only take the user memory space?